### PR TITLE
Fixes retain cycle in GitHubSearchRepositoriesViewController

### PR DESCRIPTION
**Pull request by TheAdamBorek**
I've found the retain cycle issue inside `GitHubSearchRepositoriesViewController.swift`.
The fix is pretty simple, I've just added missing `[weak self]` in 2 places inside the viewController

- [x] `./scripts/all-tests.sh` passes 👌

### Before:
zrzut ekranu 2017-02-13 07 16 23

### After:
zrzut ekranu 2017-02-13 07 51 00

### CODE REVIEW

File: RxExample/RxExample/Examples/GitHubSearchRepositories/GitHubSearchRepositoriesViewController.swift [hunk 1]
```diff
@@ -43,8 +43,9 @@ class GitHubSearchRepositoriesViewController: ViewController, UITableViewDelegat
```
Comment (kzaher) [inline RxExample/RxExample/Examples/GitHubSearchRepositories/GitHubSearchRepositoriesViewController.swift]: I think we can define local variable table here.

`let tableView = self.tableView`

File: RxExample/RxExample/Examples/GitHubSearchRepositories/GitHubSearchRepositoriesViewController.swift [hunk 2]
```diff
@@ -43,8 +43,9 @@ class GitHubSearchRepositoriesViewController: ViewController, UITableViewDelegat
 
 
         let loadNextPageTrigger = self.tableView.rx.contentOffset
-            .flatMap { _ in
-                self.tableView.isNearBottomEdge(edgeOffset: 20.0)
+            .flatMap { [weak self] _ -> Observable<Void> in
+                guard let `self` = self else { return Observable.empty() }
+                return self.tableView.isNearBottomEdge(edgeOffset: 20.0)
```
Comment (kzaher) [inline RxExample/RxExample/Examples/GitHubSearchRepositories/GitHubSearchRepositoriesViewController.swift]: ... so we can just say here `return tableView.isNearBottomEdge(edgeOffset: 20.0)` without self.

File: RxExample/RxExample/Examples/GitHubSearchRepositories/GitHubSearchRepositoriesViewController.swift [hunk 3]
```diff
@@ -80,7 +81,8 @@ class GitHubSearchRepositoriesViewController: ViewController, UITableViewDelegat
 
         // dismiss keyboard on scroll
         tableView.rx.contentOffset
-            .subscribe { _ in
+            .subscribe { [weak self] _ in
+                guard let `self` = self else { return }
                 if self.searchBar.isFirstResponder {
```
Comment (kzaher) [inline RxExample/RxExample/Examples/GitHubSearchRepositories/GitHubSearchRepositoriesViewController.swift]: We should be able to do the same here also, just use local variable.

File: RxExample/RxExample/Examples/GitHubSearchRepositories/GitHubSearchRepositoriesViewController.swift (hunk 1 above)
Comment (TheAdamBorek) [inline RxExample/RxExample/Examples/GitHubSearchRepositories/GitHubSearchRepositoriesViewController.swift]: 👍

At the beginning I wanted to put the tableView inside the capture list to the closure. However, the tableView is forced unwrapped optional and when you put it in the capture list it become the optional.

That's also a reason why I set type explicit in `let tableView: UITableView = self.tableView`

**Pull request merged**